### PR TITLE
New extended pagination

### PR DESCRIPTION
**Pull request by Shpartko**
Hi everyone!
I was create new extended pagination for my lovely tables :)
Try it now http://jsfiddle.net/n8uq78hz/1/
I hope you will like it.

### CODE REVIEW

File: docs/_i18n/en/documentation/table-options.md [hunk 1]
```diff
@@ -517,6 +517,27 @@ The table options are defined in `jQuery.fn.bootstrapTable.defaults`.
         <td>'&rsaquo;'</td>
         <td>Indicate the icon or text to be shown in the pagination detail, the next button.</td>
     </tr>
+    <tr>
+        <td>paginationSuccessivelySize</td>
+        <td>data-pagination-successively-size</td>
+        <td>Number</td>
+        <td>10</td>
```
Comment (wenzhixin) [inline docs/_i18n/en/documentation/table-options.md]: Default value is `3`

File: docs/_i18n/en/documentation/table-options.md [hunk 2]
```diff
@@ -517,6 +517,27 @@ The table options are defined in `jQuery.fn.bootstrapTable.defaults`.
...
+        <td>Maximum successively number of pages in a row.</td>
+    </tr>
+    <tr>
+        <td>paginationPagesBySide</td>
+        <td>data-pagination-pages-by-side</td>
+        <td>Number</td>
+        <td>3</td>
```
Comment (wenzhixin) [inline docs/_i18n/en/documentation/table-options.md]: Default value is `1`

File: docs/_i18n/en/documentation/table-options.md [hunk 3]
```diff
@@ -517,6 +517,27 @@ The table options are defined in `jQuery.fn.bootstrapTable.defaults`.
...
+        <td>Number of pages on each side (right, left) of the current page.</td>
+    </tr>
+    <tr>
+        <td>paginationUseIntermediate</td>
+        <td>data-pagination-use-intermediate</td>
+        <td>Boolean</td>
+        <td>true</td>
```
Comment (wenzhixin) [inline docs/_i18n/en/documentation/table-options.md]: Default value is `false`

File: src/bootstrap-table.js [hunk 4]
```diff
@@ -306,6 +306,9 @@
         paginationDetailHAlign: 'left', //right, left
         paginationPreText: '&lsaquo;',
         paginationNextText: '&rsaquo;',
+        paginationSuccessivelySize: 10, // Maximum successively number of pages in a row
+        paginationPagesBySide: 3, // Number of pages on each side (right, left) of the current page.
+        paginationUseIntermediate: true, // Calculate intermediate pages for quick access
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: The same with above.

**Comment by wenzhixin** [comment]
BTW, This branch has conflicts that must be resolved, thanks @Shpartko

**Comment by djhvscf** [comment]
Hi! We don't have any reply in that case we are going to close this PR. If you want, please create a new **PR** with this change against develop. This is an awesome feature.

**Pull request closed without merging**